### ISSUE TITLE
Tacoma faction can be angered before they spawn, and in a different location

### ISSUE THREAD

**Issue by sparr** [CleverRaven/Cataclysm-DDA#77267]
### Describe the bug

Currently the tacoma commune is not unique, and the game picks one to spawn NPCs at for the appropriate quest. However, before this happens (and also after?), all of the locations belong to their faction, so you will anger them if you kill the animals, steal the solar panels, disassemble the tractor, etc. Even though the farm is empty when you raid it.

### Attach save file

N/A

### Steps to reproduce

1. Find tacoma commune ranch
2. Observe it is empty of NPCs
3. Do something to anger their faction
4. Go to refugee center
5. Do quests to reach the tacoma commune quest
6. Go to the selected instance of the ranch
7. They are angry at you

### Expected behavior

I would expect the ranch locations to not belong to the faction until the faction spawns, and then just one such ranch to belong to them. I would probably expect them to spawn in a pristine ranch, not one the player has already raided.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: EndeavourOS; Description: EndeavourOS Linux; Release: rolling; Codename: rolling; 
- Game Version: 9e88c2a62c-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by PatrikLundell** [comment]
I don't think it's possible for the logic to detect if the player has interacted with a Tacoma farm to use that to determine which instance to use. Also, that wouldn't really make sense, because the Refugee Center should send their people to "their" instance regardless of whether it has been looted, since that's their known location.

I agree nothing should belong to any faction prior to the generation of the faction, though.

One possible way to deal with the issue might be to actually spawn a bare bones crew at THE Tacoma farm when the game starts and have this crew refer you to the Refugee Center (another bread crumb pointing in that direction) if you haven't gotten the Refugee Center mission yet.
The might say something to the effect of having been sent out to survey and prepare the site, but are now stuck waiting for further crew and orders. The might even be willing to do some trading.
However, this may have to wait until conditions exist to protect a unique Tacoma Farm instance.

**Comment by sparr** [comment]
Yeah, uniqueness and a small crew sounds great. If that's on the short term roadmap, great. If not, I'd like to try to fix this sooner.

When you say "their" instance and "known location"... Isn't that picked dynamically when the quest is generated?

**Comment by PatrikLundell** [comment]
Yes, the instance is currently picked when the quest is generated, but if the location is unique there's only one to pick from, and similarly, if only one is populated, the search could be for that one instead.

I don't believe there to be any roadmap as such. There are some goals that are used to reject development running counter to them, but there's nobody assigning tasks to people. Rather, individuals work on what they feel like, or believe is useful.

I believe there is sort of a goal to make locations that ought to be unique after means to ensure those locations are accessible and not overrun by monsters have been developed. However, I may be biased, as I argue for the development of these protection measures before making locations unique, while others are more keen on the uniqueness than ensuring their contents is accessible (i.e. different priority when it comes to the order: I don't think anyone is opposed to means for ensuring content availability eventually).

**Comment by IdleSol** [comment]
Or move the ranch to the refugee center. There's a place there. To be more specific, move the tasks and buildings. Add fence upgrades and get a fenced area where you can place people, animals or plant trees/plants. The area of the center and ranch are not much different.

The second option, after selecting the ranch, is to do a full omt upgrade. Then it doesn't matter what the player was doing there. The foreman came, brought his stuff and cleaned up the “trash” left by the player.

**Comment by PatrikLundell** [comment]
Moving Tacoma into the Refugee Center doesn't really help much. Sure, it ensures physical reachability, but it doesn't do anything to protect against monsters. The vast empty spare of the Refugee Center IS the protection currently, because it takes time in the reality bubble for monsters to move in from the surrounding areas, but that's it.

An OMT replacement would sort of work, but it's really a bummer for your new player who finds this nice place and sets up his base there, only to be invaded by goons from the Refugee Center who kick him out and steals or destroys his stuff. Although I admit this can already happen, but at least you ought to be able to get some of your stuff back.
